NEWS.md

```diff
@@ -1,6 +1,22 @@
 
 # vctrs (development version)
 
+* `vec_cbind()` now calls `vec_restore()` on inputs emptied of their
+  columns. This happens before taking the common type. This has
+  consequences for data frame classes with special columns that
+  devolve into simpler classes when the columns are subsetted
+  out. These classes are now always simplified by `vec_cbind()`.
+
+  For instance, column-binding a grouped data frame with a data frame
+  now produces a tibble (the simplified class of a grouped data
+  frame).
+
+* dplyr methods are now implemented for `vec_restore()`,
+  `vec_ptype2()`, and `vec_cast()`. The user-visible consequence (and
+  breaking change) is that combining a grouped data frame and a data
```

Comment:
By combining you mean row-binding?

Reply:
I meant any operation relying on a `vec_ptype2()` reduction, including row-binding.